cli/ping.go

```diff
@@ -150,11 +153,20 @@ func (r *RootCmd) ping() *serpent.Command {
 			diags := conn.GetPeerDiagnostics()
 			cliui.PeerDiagnostics(inv.Stdout, diags)
 
+			ni := conn.GetNetInfo()
 			connDiags := cliui.ConnDiags{
 				PingP2P:       didP2p,
 				DisableDirect: r.disableDirect,
-				LocalNetInfo:  conn.GetNetInfo(),
+				LocalNetInfo:  ni,
 			}
+
+			awsRanges, err := cliutil.FetchAWSIPRanges(ctx, cliutil.AWSIPRangesURL)
+			if err != nil {
+				_, _ = fmt.Fprintf(inv.Stdout, "Failed to retrieve AWS IP ranges: %v\n", err)
```

Comment:
Some people will be in firewalled or even air-gapped environments, and so won't be able to retrieve these ranges.

Firstly, you have a bug where we can fail to retrieve the ranges, set `awsRanges` to nil, and then segfault in `isAWSIP` later.  One way to do it is for `isAWSIP` to return false if the ranges is nil.

Secondly, if we can't retrieve the ranges, we should just log it at `debug` level so that people who don't have public internet access aren't spammed by this failure every time they run `coder ping`.